ui-lib/src/androidTest/java/co/electriccoin/zcash/ui/screen/send/integration/SendViewIntegrationTest.kt

```diff
@@ -1,69 +1,99 @@
 package co.electriccoin.zcash.ui.screen.send.integration
 
+import androidx.compose.ui.test.assertTextContains
 import androidx.compose.ui.test.junit4.StateRestorationTester
 import androidx.compose.ui.test.junit4.createComposeRule
 import androidx.compose.ui.test.onNodeWithText
-import androidx.compose.ui.test.performClick
 import androidx.test.filters.MediumTest
-import cash.z.ecc.android.sdk.model.ZecSend
+import cash.z.ecc.android.sdk.Synchronizer
+import cash.z.ecc.android.sdk.fixture.WalletFixture
+import cash.z.ecc.android.sdk.model.ZcashNetwork
+import cash.z.ecc.android.sdk.model.defaultForNetwork
+import cash.z.ecc.sdk.fixture.ZatoshiFixture
 import cash.z.ecc.sdk.fixture.ZecSendFixture
+import co.electriccoin.lightwallet.client.model.LightWalletEndpoint
 import co.electriccoin.zcash.ui.R
-import co.electriccoin.zcash.ui.screen.send.SendViewTestSetup
-import co.electriccoin.zcash.ui.screen.send.model.SendStage
+import co.electriccoin.zcash.ui.screen.send.WrapSend
+import co.electriccoin.zcash.ui.screen.send.assertOnConfirmation
+import co.electriccoin.zcash.ui.screen.send.assertOnForm
+import co.electriccoin.zcash.ui.screen.send.clickBack
+import co.electriccoin.zcash.ui.screen.send.clickCreateAndSend
+import co.electriccoin.zcash.ui.screen.send.setAddress
+import co.electriccoin.zcash.ui.screen.send.setAmount
+import co.electriccoin.zcash.ui.screen.send.setMemo
+import co.electriccoin.zcash.ui.test.getAppContext
 import co.electriccoin.zcash.ui.test.getStringResource
 import kotlinx.coroutines.runBlocking
 import org.junit.Rule
 import org.junit.Test
-import kotlin.test.assertEquals
-import kotlin.test.assertNotEquals
 
 class SendViewIntegrationTest {
 
     @get:Rule
     val composeTestRule = createComposeRule()
 
-    private fun newTestSetup(
-        sendStage: SendStage = SendStage.Form,
-        zecSend: ZecSend? = null
-    ) = SendViewTestSetup(
-        composeTestRule,
-        sendStage,
-        zecSend
+    private val wallet = WalletFixture.Alice
+    private val network = ZcashNetwork.Testnet
+    private val spendingKey = runBlocking {
+        WalletFixture.Alice.getUnifiedSpendingKey(
+            seed = wallet.seedPhrase,
+            network = network
+        )
+    }
+    private val synchronizer = Synchronizer.newBlocking(
```

Comment:
Seeing the synchronizer here, I realized a limitation of my prior suggestion.  Synchronizer performs network access and disk IO, making it not great for reproducible or isolated tests.  One option would be to move this to the integration-test-lib, but I don't think that solves all the problems.

A better option might be MockSynchronizer that throws exceptions except for the `send` method which we can then return a fake result flow.  That fake result flow makes it easier for us to also test out moving from the different states (e.g. success/failure)

Reply:
Yep, that is a good point too! Thanks. I prepared the mocked Synchronizer, which could have a big potential for our tests, unfortunately for this use-case, there is a limitation of the PendingTransaction visibility. But we still can have the send method returning an empty flow, which is sufficient here.